### DISCUSSION TITLE
Jamulus.app would like to record this computers screen

### DISCUSSION THREAD

**Discussion by BigScottRocks** [jamulussoftware/jamulus#2885] [Technical Support & Help]
Good Morning, 

BigScottRocks from the IP Rockers, not a newbie to Jamulus, been using it for about three years now. Last night I had this dialogue box popup on my iMac. Nothing new was installed on iMac, and I have been running this version of Jamulus since it was released. The only thing I can say I did different was  Jam on this public server, I normally jam in Private.  Here's the details.

iMac running macOS Monterey version 12.6
Jamulus version 3.9 
Qt version 6.3.1

![IMG_3088](https://user-images.githubusercontent.com/96243817/192545648-5e51af04-d730-4fec-9fed-c428ec744483.jpg)

Any ideas on this one. First time I've ever seen this message.  I clicked on "deny"" and continued to jam.

**Comment by pljones** [comment]
>  I clicked on "deny"" and continued to jam.

I'd click "deny" on it, too.  I'm not a Mac user and I've no idea why that permission would be needed by Jamulus.

It's certainly somewhat worrying.  @ann0see or @emlynmac, can you shed any light on it?

**Reply by emlynmac** [reply]
No clue. My guess would be something in QT is now triggering the permissions request.

**Reply by ann0see** [reply]
No. Actually I don’t know why this comes up. It might be related to the accessibility feature (audio alerts?) otherwise, could you please send us a hash from the app? Hopefully it’s not compromised.

**Comment by BigScottRocks** [comment]
Here you go. Please let me know if you need anything else to diagnose.  

bigscottrocks@Scotts-iMac Downloads % md5 jamulus_3.9.0_mac.dmg 
MD5 (jamulus_3.9.0_mac.dmg) = 0a7eaceb5fd72576fade047ddeb32775

bigscottrocks@Scotts-iMac ~ % md5 /Applications/Jamulus.app/Contents/MacOS/Jamulus
MD5 (/Applications/Jamulus.app/Contents/MacOS/Jamulus) = a4e6cb8e9c034928fcc4405e285bb3b2

**Reply by rdica** [reply]
I went looking for, but could not find posted md5 or shasums of the tarballs and pkgs.

**Reply by hoffie** [reply]
> bigscottrocks@Scotts-iMac Downloads % md5 jamulus_3.9.0_mac.dmg
> MD5 (jamulus_3.9.0_mac.dmg) = 0a7eaceb5fd72576fade047ddeb32775
> 
> bigscottrocks@Scotts-iMac ~ % md5 /Applications/Jamulus.app/Contents/MacOS/Jamulus
> MD5 (/Applications/Jamulus.app/Contents/MacOS/Jamulus) = a4e6cb8e9c034928fcc4405e285bb3b2

I can confirm both these hashes (by downloading and md5sum'ing the (extracted) files.

**Reply by ann0see** [reply]
> I went looking for, but could not find posted md5 or shasums of the tarballs and pkgs.

We don't publish them at the moment. macOS should be ad hoc signed and not open if there's a hash mismatch.

**Comment by hoffie** [comment]
- Can you clarify when exactly that dialog appeared? Right after connecting? At some later point?
- Was it upon receiving a chat message by chance?
- Do you have accessibility features enabled in your Jamulus settings?
- Can you reproduce this issue?

I have a hard time coming up with ideas about possible reasons. The Qt6 switch might be a clue, but I don't think it even has a Mac permission API at all, so if it does cause it, it does so via some internal mechanism.
I haven't found any hints about Mac-side bugs around such dialogs.

I've also checked the `.plist` files and don't see any relevant references. I haven't found permission requesting code in the `qtmultimedia` module either.

**Comment by BigScottRocks** [comment]
See below answers to your questions.

Can you clarify when exactly that dialog appeared? Right after connecting? At some later point?  I had been jamming for at least 15/220 minute when the message popped up.
Was it upon receiving a chat message by chance? I was not watching chat, so not sure on this one.
Do you have accessibility features enabled in your Jamulus settings?  I do not have accessibility options on 
Can you reproduce this issue?  I tried by joining same server, had someone type in chat, but thus far I cannot reproduce.

**Comment by dcorson-ticino-com** [comment]
This is scary.
Jamulus never tries to record a screen shot as far as I know and I don't see any reason why it would want to ever.
Have you done a virus scan of your compi?
Don

**Reply by BigScottRocks** [reply]
Yes, computer is clean. I run Symantec End Point and its up to date.

**Reply by ann0see** [reply]
Hmm, I think I vaguely remember getting a message like that around the time I tested the accessibility improvements with audio alerts and somehow expected it. @chigkim do you remember anything?

@BigScottRocks Can you reset Jamulus' screen record permissions (https://apple.stackexchange.com/questions/384230/how-do-i-reset-screen-recording-permission-on-macos-catalina#384231)?

**Reply by richard-dicaire-angi** [reply]
To get bundle ID:
https://stackoverflow.com/questions/39464668/how-to-get-bundle-id-of-mac-application

`mdls -name kMDItemCFBundleIdentifier -r /Applications/Jamulus.app`
or
`osascript -e 'id of app "Jamulus.app"'`
both of which return: `io.jamulus.Jamulus`

**Reply by BigScottRocks** [reply]
I have reset Jamulus' screen record permissions.  Will wait till the popup happens again.

Successfully reset ScreenCapture approval status for io.jamulus.Jamulus

**Reply by ann0see** [reply]
> Successfully reset ScreenCapture approval status for io.jamulus.Jamulus

Great. If it shows up again, please write here.

**Comment by ann0see** [comment]
Maybe related https://github.com/keepassxreboot/keepassxc/issues/5224

https://github.com/keepassxreboot/keepassxc/issues/3609

**Comment by chigkim** [comment]
That's strange. I never got a notification about screen recording for 
Jamulus.

  Why would Jamulus need to record screen?

**Reply by ann0see** [reply]
>  Why would Jamulus need to record screen?

I assume it's not a screen recording but a permission needed which you only get if you request screen recording permissions.

**Comment by chigkim** [comment]
As far as I know, accessibility features should not trigger screen recording permissions, and I never got one before.

**Comment by ann0see** [comment]
I think I could now reproduce it: right click on the mixerboard and on what's this. This gives "Jamulus would like to record your screen" message!

**Reply by pljones** [reply]
Whu...?????!

**Comment by BigScottRocks** [comment]
I was able to reproduce the screen recording dialogue box. I don't see anyway to attach the new screenshot I grabbed but here is how to reproduce.  **Right click** anywhere within the main jamulus mixer window. You will get one option, "**what's this**" Click on "what's this" , it will generate the screen recording security dialogue box.

**Reply by pljones** [reply]
Only when connected to a server or even offline?

Only on the mixer?  Nowhere else for "What's this?"?

And, for others, we've had this on:
> iMac running macOS Monterey version 12.6
> Jamulus version 3.9
> Qt version 6.3.1

@ann0see, what version did you confirm on?

We would appreciate a 👍 from anyone with the _same_ set up as above to confirm and a comment from anyone with a different set up _also able to reproduce_ (plus 👍 from anyone able to confirm those).  Thanks.

**Reply by BigScottRocks** [reply]
You can produce it if you're connected to a server or disconnected,  Either way. I did it both ways.

**Reply by ann0see** [reply]
> what version did you confirm on?

3.9.1 also on Monterey (I think that's what runs on my VM)

**Reply by hoffie** [reply]
Some further questions to those who can reproduce:
1. Is there someone on a Mac who can*not* reproduce this issue using the above instructions? If so, please post here.
2. Does it matter where you right click? In other words: Can you reproduce this issue with two different What's this texts (e.g. input level faders)?
3. What language are you using? Can you try with a different UI language?
4. Is there a previous Jamulus version which did not behave that way? 3.8.x would be interesting to test.

The English What's this text contains the word "recording" and I'm wondering whether it is related to it (although it'd be really, really strange).

**Reply by ann0see** [reply]
~~3.7.0 also shows this on macOS 12.6, 3.5.12 doesn't seem to. But 3.5.12 might not. 3.6.0 doesn't either, and 3.6.2 doesn't either.~~

Ignore

**Reply by ann0see** [reply]
Hmm. Now I can't reproduce it anymore with 3.7.0, 3.8.0... (reset the permissions). Maybe it was an old message I didn't close.

... Ignore. It does happen. Also for 3.5.12. I didn't do the correct procedure.

**Reply by ann0see** [reply]
Ok. Even 3.4.3 shows this issue.

3.3.6 too.

Edit: Even the first supported version by Monteray (3.3.5) shows this issue.

**Reply by pljones** [reply]
:roll_eyes: I'm glad "What's this?" is so widely used...  Oh well.

Still, if it's reliably triggered, it should make pinning down the trigger easier - i.e. why it happens for that widget rather than others, for a start.

**Reply by ann0see** [reply]
The question is if it’s only one widget. I doubt that it’s only the mixer board.

**Comment by chrisrimple** [comment]
CAN REPRO using Jamulus 3.8.1 on Mac M1 running macOS Monterey 12.6.  CANNOT REPRO using Jamulus 3.8.1 on Mac Intel running macOS Mojave 10.14.6.

**Comment by Rob-NY** [comment]
Reproduced on Jamulus 3.8.2 (QT 5.15.2) on Mac Mini running Big Sur (11.7).   Once you deny it will not pop up again unless cleared by going to Settings->Security & Privacy->Screen Recording (unlock for changes) and remove the Jamulus entry entirely.  I can reproduce this anywhere "what's this" appears -- not just on the mixerboard.   Got it when clicking what's this on Reverb, Jitter literal, Delay literal, etc.

**Comment by ann0see** [comment]
Thanks @Rob-NY @chrisrimple 
I'd conclude that newer OS show this behaviour regardless of the Jamulus version. It might be a Qt bug.

**Reply by hoffie** [reply]
Given that the behavior affects both Qt5 and Qt6 builds but seems to have been introduced by a Mac version change, I'd rather suspect a strange interaction here. Maybe we can find another Qt application to try in order to rule out anything Jamulus-specific?
